src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs

```diff
@@ -38,37 +55,241 @@ public class PackageMetadataResourceV3 : PackageMetadataResource
             Common.ILogger log,
             CancellationToken token)
         {
-            var metadataCache = new MetadataReferenceCache();
-
-            var packages =
-                (await _regResource.GetPackageMetadata(packageId, includePrerelease, includeUnlisted, sourceCacheContext, log, token))
-                    .Select(ParseMetadata)
-                    .Select(m => metadataCache.GetObject(m))
-                    .ToArray();
+            var range = VersionRange.All; // We get all versions for package and later filter down the road get ones we need.
```

Comment:
```suggestion
            var range = VersionRange.All;
```

There's no filtering happening any more.

Reply:
Removed.